tf_encrypted/keras/engine/base_layer.py

```diff
@@ -115,17 +115,15 @@ def set_weights(self, weights, sess=None):
           matching the output of layer.get_weights()
       sess: tfe session"""
 
-    # Define keras weights as private variable
-    tfe_weights_pl = [tfe.define_private_placeholder(w.shape)
-                      for w in weights]
-
     # Assign new keras weights to existing weights defined by
     # default when tfe layer was instantiated
     if not sess:
       sess = K.get_session()
     for i, w in enumerate(self.weights):
-      fd = tfe_weights_pl[i].feed(weights[i])
-      sess.run(tfe.assign(w, tfe_weights_pl[i]), feed_dict=fd)
+      shape = w.shape.as_list()
+      tfe_weights_pl = tfe.define_private_placeholder(shape)
+      fd = tfe_weights_pl.feed(weights[i].reshape(shape))
+      sess.run(tfe.assign(w, tfe_weights_pl), feed_dict=fd)
```

Comment:
I had to change a bit the logic here so the placeholder shape gets defined based on the layer shape weights instead of numpy arrays from `get_weights`. It was causing some issue for the bias in the conv2d. The shape of the numpy array is rank 1 [filter], when the conv2d layer needs a bias of rank=3

Reply:
I'm not sure if this helps, but I had imagined this being two separate sets of logic -- one for numpy arrays and one for TF Variables.